### ISSUE TITLE
[no-misused-promises] The rule does not take effect when the async function is called without await

### ISSUE THREAD

**Issue by CaptainOfPhB** [typescript-eslint/typescript-eslint#3173]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-misused-promises": ["error"]
  }
}
```

```TS
async function getMarkdownUrlOfCurrentTab(): Promise<string> {
  const [tab] = await chrome.tabs.query({ active: true, currentWindow: true });
  const { url, title } = tab;
  return `[${title}](${url})`;
}

function copyUrl() {
  const text = getMarkdownUrlOfCurrentTab();
              ^^^^
  // expected 'string', but got 'Promise<string>' because missed await, but eslint has not hint/error
  console.log(text);
}
```

**Expected Result**

eslint can hint/output error: missed await

**Actual Result**

eslint has no hint/error

**Additional Info**

I ran `eslint --fix src --ext .ts,.tsx` command, but nothing happend. Below repository can reproduce this issue:

Repo: [https://github.com/CaptainOfPhB/yuque-plugin](https://github.com/CaptainOfPhB/yuque-plugin/tree/typescript-eslint/no-misused-promises)
Branch: [typescript-eslint/no-misused-promises](https://github.com/CaptainOfPhB/yuque-plugin/tree/typescript-eslint/no-misused-promises)
File: [src/actions/copyUrl.ts#L8](https://github.com/CaptainOfPhB/yuque-plugin/blob/typescript-eslint/no-misused-promises/src/actions/copyUrl.ts#L8)

**Versions & Env**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.17.0` |
| `@typescript-eslint/parser`        | `^4.17.0` |
| `TypeScript`                       | `^4.2.3` |
| `ESLint`                           | `^7.21.0` |
| `node`                             | `v15.5.1` |

| IDE                            | version |
| ---------------------------------- | ------- |
| WebStorm 2021.1 EAP | Build #WS-211.6305.8, built on Merch 2, 2021 |

**Comment by bradzacher** [comment]
there's nothing wrong with this code.

`const text = awaitablefunction()` is completely valid code.
because you can do `await text` or `Promise.all([text])` or something of the sort to await the promise later.

Similarly - passing the promise into `console.log` is valid. `console.log` does not expect a `string` argument - it specifically expects anything.

**Comment by CaptainOfPhB** [comment]
Thanks for your reply at first!

Of course I know those code are valid, but is not what I expected. According to the **correct** code in the document of [no-misused-promise](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-promises.md):

```js
// Examples of correct code for this rule:

const promise = Promise.resolve('value');

if (await promise) {
  // Do something
}

const val = (await promise) ? 123 : 456;

while (await promise) {
  // Do something
}

for (const value of [1, 2, 3]) {
  await doSomething(value);
}

new Promise((resolve, reject) => {
  // Do something
  resolve();
});

const eventEmitter = new EventEmitter();
eventEmitter.on('some-event', () => {
  doSomething();
});
```

If eslint can not hint me maybe I missed await, my code will cause unexpected errors because I often forget to add await. So, my question is: **why eslint cannot hint me maybe I missed await even if I add "no-misused-promises" rule in eslint configuration file** or **how can I config eslint, so eslint can hint me maybe I missed await**.

Thanks again!

**Comment by bradzacher** [comment]
The lint rule only looks for common misuses of promises places were we can tell that you likely forgot to use an await.

If you just assign a promise to a variable ***it is not a misused promise***

It's completely correct code to do this.

Similarly passing a promise to `console.log` is valid. Doing so lets you see the state of the promise whilst developing, which can be helpful in figuring out what state your system is in.

If you want a log function that does not accept any value - then make one with stricter types

**Comment by CaptainOfPhB** [comment]
I got it, Thanks for you reply!
